tools/wpt/testfiles.py

```diff
@@ -183,24 +181,12 @@ def _in_repo_root(full_path):
     return len(path_components) < 2
 
 
-def _init_manifest_cache():
-    c = {}
-
-    def load(manifest_path=None, manifest_update=True):
-        if manifest_path is None:
-            manifest_path = os.path.join(wpt_root, "MANIFEST.json")
-        if c.get(manifest_path):
-            return c[manifest_path]
-        # cache at most one path:manifest
-        c.clear()
-        wpt_manifest = manifest.load_and_update(wpt_root, manifest_path, "/",
-                                                update=manifest_update)
-        c[manifest_path] = wpt_manifest
-        return c[manifest_path]
-    return load
-
-
-load_manifest = _init_manifest_cache()
+def load_manifest(manifest_path=None, manifest_update=True):
+    from manifest import manifest
```

Comment:
Switch to a delayed import here because otherwise Python will think there are two different `manifest` modules (because of different import paths) and we don't get the singleton.

Reply:
What do you say we document your rationale with a comment? I'm afraid of your
optimization being accidentally subverted by some future refactoring (the
corresponding performance degradation might not be noticed immediately).

This is beyond my experience with Python, though. Can you advise on any
longer-term goals that would obviate this kind of consideration?

Reply:
I added a comment here, but I don't really have great ideas how to make sure we always get a singleton.

Reply:
Did you verify that the previous code was broken? I know that `foo.bar` and `bar` are different even if they're the same `bar.py` file, but I'm slightly surprised if that also affects relative imports.